lib/valueflow.cpp

```diff
@@ -524,6 +524,15 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
         }
     }
 
+    else if (parent->str() == "?" && tok == parent->astOperand1() && value.isKnown() && parent->astOperand2() && parent->astOperand2()->astOperand1() && parent->astOperand2()->astOperand2()) {
```

Comment:
I have the feeling you should check that the value is an int also.

Reply:
Indeed, I missed that. Thanks !